src/Microsoft.Health.Fhir.SqlServer/Features/Search/Expressions/Visitors/QueryGenerators/SqlQueryGenerator.cs

```diff
@@ -353,7 +352,7 @@ public object VisitTable(TableExpression tableExpression, SearchOptions context)
                     {
                         // In case its revinclude, we limit the number of returned items as the resultset size is potentially
                         // unbounded. we ask for +1 so in the limit expression we know if to mark at truncated...
-                        StringBuilder.Append("TOP (").Append(Parameters.AddParameter(MaxIncludedItems + 1)).Append(") ");
+                        StringBuilder.Append("TOP (").Append(Parameters.AddParameter(context.MaxItemCount + 1)).Append(") ");
```

Comment:
MaxItems when relating to _include_ or _revInclude_ is not the same as the number of items on the page. See: http://hl7.org/fhir/http.html#paging. E.g. you may want 10 patients (the page size) _and_ include up to n (in this case 100) of their observation records. Likely you'd need a separate configuration for this. This is one of those slightly unexpected parts of the FHIR spec.

Reply:
I see. Let me revert it to original code.
Do you want me to create another issue, for it, or extend current PR and introduce another option?

---
In reply to: [471918805](https://github.com/microsoft/fhir-server/pull/1195#discussion_r471918805) [](ancestors = 471918805)

Reply:
Since we now have the other two, I think having a separate option for this would be good

Reply:
Added.
I'm looking into build failures, but they test works locally, so I'm not sure what is going on.

---
In reply to: [473172836](https://github.com/microsoft/fhir-server/pull/1195#discussion_r473172836) [](ancestors = 473172836)